back-end/src/base/calls/calls.repository.ts

```diff
@@ -17,10 +26,176 @@ export class CallsRepository {
 		return this.db.collection<Course>('courses');
 	}
 
+	async isWeekEnd(date: Date) {
+		const day = date.getDay();
+		return day === 0 || day === 6;
+	}
+	async checkWeekEnd() {
+		const date = new Date();
+		const isWeekEnd = await this.isWeekEnd(date);
+		if (isWeekEnd) {
+			throw new ServiceError('FORBIDDEN', 'Vous ne pouvez pas performer ces actions le week-end');
+		}
+	}
+
+	async getActualCourses() {
+		const actualDate = new Date();
+
+		return await this.courses
+			.find({
+				$and: [{ periodStart: { $lte: actualDate } }, { periodEnd: { $gte: actualDate } }],
+			})
+			.toArray();
+	}
+
+	async getClassSupervisor(courseId: InferIdType<Course>) {
+		const courseObjectId = new ObjectId(courseId);
+		const course = await this.courses.findOne({ _id: courseObjectId });
+		// @ts-ignore (Mandatory because of typo issue)
```

Comment:
Your id should be an objectId in the database btw.
And i just verified, it is.
So you don't need to convert it when it's already one.

Reply:
Already fixed

Reply:
Okay, why is the ts-ignore still there ?

Reply:
Nevermind i checked.
You have a missmatch of data from what's in the database courses table and what's in the interface of your collection.

Reply:
Okay it's fixed thanks for the advice